scripts/flag_reviews.py

```diff
@@ -1,66 +1,69 @@
-                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
-                    row['flag'] = 'Duplicate'
+                if (safety_info == duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower()) and (useful_info == duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower() and pd.notnull(row['openingTime']) and pd.notnull(row['closingTime'])):
+                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
+                    df.at[index , 'flag'] = Flag.DUPLICATE
                     continue                    
                 # else: 
-                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
+                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
 
         else:
             duplicate_reviews_dict[dict_key] = row
 
-        if not row['Latitude'] or not row['Longitude']:
-            row['flag'] = 'Empty lat/long'
+        if not row['latitude'] or not row['longitude'] or np.isnan(row['latitude']) or np.isnan(row['longitude']):
+            df.at[index , 'flag'] = str(Flag.MISSING_LAT_LONG)
             continue
             
-        if store_name.isdigit() or (row['Address'] and store_name == row['Address'][:len(store_name)]):
-            row['flag'] = 'Invalid store name'
+        if store_name.isdigit() or (row['address'] and store_name == row['address'][:len(store_name)]):
+            df.at[index , 'flag'] = Flag.INVALID_STORE_NAME
             continue
 
         if len(useful_info) < 5 and len(safety_info) < 5:
-            row['flag'] = 'Very short reviews'
+            df.at[index , 'flag'] = Flag.VERY_SHORT_REVIEW
             continue
 
         if len(useful_info) < 10 and len(safety_info) < 10:
-            row['flag'] = 'Short reviews'
+            df.at[index , 'flag'] = Flag.SHORT_REVIEW
             continue
 
         if (useful_info and len(set(useful_info)) < 4) or (safety_info and len(set(safety_info)) < 4):
-            row['flag'] = 'Spammy words'
+            df.at[index , 'flag'] = Flag.SPAMMY_REVIEW
 
-    # df = df.drop(columns=['dt_Timestamp'])
+    df = df.drop(columns=['updatedAt'])
+    # Convert NaN to None as JSON cannot serialise with NaN type
+    df = df.where(pd.notnull(df), None)
+    # Only display non-empty flags
+    df = df[df['flag']!= '']
```

Comment:
I suspect Key Errors here, Could we do some error handling here?

Reply:
hmm, good catch. i generally prefer `dict.get(key)` instead of `dict[key]` since get catches KeyErrors. ideally, the df object should be defined somewhere (like in a class or something), and keys should be attributes. that way key changes would need to be done only once.

Reply:
Agreed, will make those changes in the next iteration.